File: src/main/java/testla/screenplay/question/Question.java
package testla.screenplay.question;

import testla.screenplay.actor.IActor;

/**
 * Questions can be triggered by calling them from an actor object.
 *
 * @author Patrick Döring
 */
public abstract class Question<T> implements IQuestion<T>{
    /**
     * Implementation of the query answer.
     *
     * @param actor the actor that queries.
     * @return Object.
     */
    public abstract T answeredBy(IActor actor);
}

File: src/main/kotlin/de/p7s1/testla/screenplay/Question.kt
package de.p7s1.testla.screenplay
import de.p7s1.testla.IQuestion
import de.p7s1.testla.IActor

abstract class Question<T>: IQuestion<T> {
    /**
     * Implementation of the query answer.
     *
     * @param actor the actor that queries.
     * @returns Any
     */
    abstract override fun answeredBy(actor: IActor): T
}
